File: tezocare_mobile/lib/features/visit/domain/entities/vitals.dart
import 'package:equatable/equatable.dart';

class Vitals extends Equatable {
  final int? id;
  final int? visitId;
  final double? temperature;
  final int? heartRate;
  final int? respiratoryRate;
  final double? weight;
  final String? mucousMembranes;
  final int? capillaryRefillTime;
  final String? hydrationStatus;
  final String? otherFindings;
  final DateTime? recordedAt;

  const Vitals({
    this.id,
    this.visitId,
    this.temperature,
    this.heartRate,
    this.respiratoryRate,
    this.weight,
    this.mucousMembranes,
    this.capillaryRefillTime,
    this.hydrationStatus,
    this.otherFindings,
    this.recordedAt,
  });

  @override
  List<Object?> get props => [
        id,
        visitId,
        temperature,
        heartRate,
        respiratoryRate,
        weight,
        mucousMembranes,
        capillaryRefillTime,
        hydrationStatus,
        otherFindings,
        recordedAt,
      ];
}

File: backend/app/models/vitals.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, Integer, Float, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.core.database import Base


class Vitals(Base):
    __tablename__ = "vitals"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    visit_id = Column(UUID(as_uuid=True), ForeignKey("visits.id"), nullable=False)
    blood_pressure_systolic = Column(Integer, nullable=True)
    blood_pressure_diastolic = Column(Integer, nullable=True)
    glucose = Column(Float, nullable=True)
    temperature = Column(Float, nullable=True)
    weight = Column(Float, nullable=True)
    heart_rate = Column(Integer, nullable=True)
    spo2 = Column(Integer, nullable=True)
    recorded_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

    visit = relationship("Visit", back_populates="vitals")
